src/steps/index.test.ts

```diff
@@ -133,3 +133,140 @@ test('should collect data', async () => {
     },
   });
 });
+
+test('stopFetchingRequests', () => {
+  const requests: AtSpokeRequest[] = [];
+  let lastExecutionTime = 0;
+  expect(stopFetchingRequests(requests, lastExecutionTime)).toBe(true);
+
+  const fakeRequest: AtSpokeRequest = {
+    subject: 'Tough stuff',
+    requester: 'Somebody',
+    owner: 'Nobody',
+    status: 'OPEN',
+    privacyLevel: 'high',
+    team: 'winners',
+    org: 'Team Co.',
+    permalink: 'not applicable',
+    id: '333333333333',
+    isAutoResolve: false,
+    isFiled: true,
+    email: 'help@help.com',
+    createdAt: new Date().toDateString(),
+    updatedAt: new Date().toDateString(),
+  };
+  requests.push(fakeRequest);
+  //last request is not older than 14 days or exTime
+  expect(stopFetchingRequests(requests, lastExecutionTime)).toBe(false);
+
+  const fakeRequest2: AtSpokeRequest = {
+    subject: 'Tough stuff',
+    requester: 'Somebody',
+    owner: 'Nobody',
+    status: 'OPEN',
+    privacyLevel: 'high',
+    team: 'winners',
+    org: 'Team Co.',
+    permalink: 'not applicable',
+    id: '333333333333',
+    isAutoResolve: false,
+    isFiled: true,
+    email: 'help@help.com',
+    createdAt: new Date().toDateString(),
+    updatedAt: '',
+  };
+  requests.push(fakeRequest2);
+  //last request has a falsy updated time, we should keep going
```

Comment:
It seems that we should probably stop on the first page if there is no other means of stopping. 🤔

Reply:
Cool. Changed that behavior. Also realized my original test wasn't checking for non-Date()-parsable strings in updatedAt, so adjusted code to handle that properly.